### PR TITLE
Friends index ps

### LINKED ISSUE #27: Friends index page
### User Story
As a registered user
When I visit the friends index page
I see a search box to add friend's by their email
I see a list of my friends

#### Extension
I see stats about my friends such as:

- best book buddy

- newest book buddy

- First book buddy

### PR DESCRIPTION
**Pull request by Strompy**
# Description
Creates Friends Index Page with friends list and friend search box
Friends list tested, search and add friends not yet implemented or tested
Creates `User::FriendsController`
Creates user/friends/index view
Creates `friendships` table and `Friend` model
Adds friends relations to `User` model
Schema and migration for that change
Model relations all tested

Broke user story up into smaller stories, book covers and add a friend are now their own issues
HTML is minimal to not get in the way of future styling

# Closes issue(s)
closes #27

# How to test / reproduce
`rspec spec/features/friends`
`rspec spec/models`

# Screenshots

# Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist
- [x] I have tested this code
- [ ] I have updated the Readme

# Other comments

### CODE REVIEW

**Comment by Strompy** [comment]
Travis failed trying to compare DateTime objects. I can't reproduce the error in my local env.
Screen Shot 2020-09-12 at 4 07 00 PM

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -27,6 +32,5 @@
     expect(new_user.token).to eq('abcdefg12345')
     expect(new_user.refresh_token).to eq('12345abcdefg')
     expect(new_user.oauth_expires_at).to eq(auth[:credentials][:expires_at])
```
Comment (ajtran303) [inline spec/models/user_spec.rb:34]: Let's comment out this assertion.

File: spec/features/navigation/nav_bar_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      expect(page).to have_link("Home", href: root_path)
+      expect(page).to have_link("Dashboard", href: user_dashboard_path)
+      expect(page).to have_link("Find Books", href: books_path)
+      expect(page).to have_link("Friends", href: user_friends_path)
+      expect(page).to have_link("My Books", href: user_books_path)
+      expect(page).to have_link("Account Info", href: user_account_path)
+    end
+  end
+  it "has working paths for all links" do
+    skip
```
Comment (ajtran303) [inline spec/features/navigation/nav_bar_spec.rb:17]: On line 17

You can pass an argument to `skip` to change the message in the terminal that says `no reason given`.

```
skip "Routes don't exist yet"
```

**Comment by Strompy** [comment]
Corrected the path helper in the application.html and commented out the time test that was failing

File: spec/features/navigation/nav_bar_spec.rb (hunk 2 above)
Comment (Strompy) [reply spec/features/navigation/nav_bar_spec.rb]: That's cool. I added that in now

File: spec/models/user_spec.rb (hunk 1 above)
Comment (Strompy) [reply spec/models/user_spec.rb]: Done

**Pull request merged**